pkg/telemetry/telemetry.go

```diff
@@ -0,0 +1,208 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package telemetry
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/cloud-on-k8s/pkg/about"
+	apmv1 "github.com/elastic/cloud-on-k8s/pkg/apis/apm/v1"
+	beatv1beta1 "github.com/elastic/cloud-on-k8s/pkg/apis/beat/v1beta1"
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	entv1beta1 "github.com/elastic/cloud-on-k8s/pkg/apis/enterprisesearch/v1beta1"
+	kbv1 "github.com/elastic/cloud-on-k8s/pkg/apis/kibana/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/kibana"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/ghodss/yaml"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	updateInterval = 10 * time.Second
+	resourceCount  = "resource_count"
+	podCount       = "pod_count"
+)
+
+var log = logf.Log.WithName("usage")
+
+type getUsageFn func(k8s.Client, []string) (string, interface{}, error)
+
+// ECK is a helper struct to marshal telemetry information.
+type ECK struct {
+	ECK   about.OperatorInfo     `json:"eck"`
+	Usage map[string]interface{} `json:"eck_usage"`
```

Comment:
Should this struct be called `Telemetry` instead? Should `Usage` be `Stats` (a term which seems be used in ES and other stack applications)? Should we have a hierarchy like so: 
`eck.build.date: "blah"` and `eck.stats.apms.pod_count: 1`

Reply:
Updated issue description.